src/field_theory/krull_topology.lean

```diff
@@ -283,3 +285,242 @@ begin
 end
 
 end totally_disconnected
+
+section compact
+
+variables {K L : Type*} [field K] [field L] [algebra K L] {E F : intermediate_field K L}
+
+/-- Let `E` be an intermediate field of `L/K` with `E/K` finite, and let `f` be an ultrafilter
+on `L ≃ₐ[K] L`. Then the restriction map `(L ≃ₐ[K] L) → (E →ₐ[K] L)` pushes `f` forward to an
+ultrafilter on `E →ₐ[K] L`. Since `E →ₐ[K] L` is a finite set, this ultrafilter is principal. The
+element of `E →ₐ[K] L` generating this principal ultrafilter is `f.alg_hom h_findim`, where
+`h_findim : finite_dimensional K E`. -/
+protected noncomputable def ultrafilter.generator_of_pushforward
```

Comment:
The next ~10 names all start with `ultrafilter.`. Please use `namespace ultrafilter` to structure the code.